packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -192,6 +206,22 @@ const ContextMenuRootContent = React.forwardRef((props, forwardedRef) => {
       side="right"
       sideOffset={2}
       align="start"
+      onCloseAutoFocus={(event) => {
+        props.onCloseAutoFocus?.(event);
+
+        if (!event.defaultPrevented && shouldPreventFocusRef.current) {
+          event.preventDefault();
+        }
+
+        shouldPreventFocusRef.current = false;
+      }}
```

Comment:
@andy-hook I saw your comment about further discussing in the `modality` branch but i find it's easier to follow if we keep all discussion related to this change in one place.

So, on this one, curious why we no longer focus the trigger here when we prevent default? We had to do it explicitly in `main` to fix `autoFocus` issues.

Reply:
This change was to match the non-modal behaviour we have in the other components. That is that when clicking outside in non-modal, focus shouldn't be returned.

Not familiar with the issue you mentioned, do you have a link to it? if this will cause issues would be good to consider it.

@benoitgrelard probably has a better explanation as he originally brought this to my attention.

Reply:
Oh wait, this is `ContextMenu` and they can't `autoFocus` a menu item so it's okay to fallback onto `FocusScope`s auto focus behaviour in this case :+1:

This was the original `autoFocus` issue if you're curious https://github.com/radix-ui/primitives/issues/518. I went through all our components and explicitly focused the trigger on unmount but looks like I also decided against applying it to `ContextMenu` at the time.